crates/provider/src/provider/trait.rs

```diff
@@ -969,12 +965,8 @@ pub trait Provider<N: Network = Ethereum>: Send + Sync {
     /// # }
     /// ```
     #[cfg(feature = "pubsub")]
-    async fn subscribe_pending_transactions(
-        &self,
-    ) -> TransportResult<alloy_pubsub::Subscription<B256>> {
-        self.root().pubsub_frontend()?;
-        let id = self.client().request("eth_subscribe", ("newPendingTransactions",)).await?;
-        self.root().get_subscription(id).await
+    fn subscribe_pending_transactions(&self) -> GetSubscription<(&'static str,), B256, N> {
+        GetSubscription::new(self.root(), "eth_subscribe", Some(("newPendingTransactions",)))
```

Comment:
we can replace the string with 

https://github.com/alloy-rs/alloy/blob/ee7a19be25c07df4898deceb9ca5d79a70e124ab/crates/rpc-types-eth/src/pubsub.rs#L76-L76